trillian/migrillian/util.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/tls"
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/trillian"
+)
+
+func toLogEntry(index int64, entry *ct.LeafEntry) (*ct.LogEntry, error) {
+	logEntry, err := ct.LogEntryFromLeaf(index, entry)
+	if _, ok := err.(x509.NonFatalErrors); !ok && err != nil {
+		return nil, fmt.Errorf("failed to parse [pre-]certificate: %v", err)
+	}
+	return logEntry, nil
+}
+
+func buildLogLeaf(entry *ct.LogEntry) (*trillian.LogLeaf, error) {
```

Comment:
There's no handy (or easily refactorable) bit of CTFE which could be reused to do the bulk of this work? That'd be nice in that it'd give ongoing confidence that this has exactly the same behaviour as CTFE does.

Reply:
I based this code on some similar functions in [trillian/ctfe/handlers.go]( https://github.com/google/certificate-transparency-go/blob/master/trillian/ctfe/handlers.go#L821). I will try to refactor it a bit to reuse.

Reply:
Cleaned this up a bit.